color-scheme should not propagate to viewport background if set on <body>

nt1m opened the pull request:
#### cbdcbdd4e8f1b5144374bef91ba9e45483f6072f
<pre>
color-scheme should not propagate to viewport background if set on &lt;body&gt;
<a href="https://bugs.webkit.org/show_bug.cgi?id=243757">https://bugs.webkit.org/show_bug.cgi?id=243757</a>

Reviewed by Simon Fraser.

Change to web compatible behavior where setting the color-scheme on &lt;body&gt; no longer propagates it to &lt;html&gt; tag for scrollbar and
webpage background purposes. This change puts WebKit in-line with Gecko &amp; Blink.

* LayoutTests/css-dark-mode/color-scheme-scrollbar-expected.txt:
* LayoutTests/css-dark-mode/color-scheme-scrollbar.html:
* Source/WebCore/page/FrameView.cpp:
(WebCore::FrameView::rendererForColorScheme const):

Canonical link: <a href="https://commits.webkit.org/253565@main">https://commits.webkit.org/253565@main</a>
</pre>










<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cab2e637a9dcf5d7005d2e4a058f3da098b004e7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/86363 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/30283 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/17332 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/95207 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/148919 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/90347 "Passed tests") | [✅ ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/64/builds/28728 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/25295 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/78499 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/90459 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/91963 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/23238 "Passed tests") | [✅ ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/73357 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/23337 "Passed tests") | 
| | [✅ ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/78248 "Build was cancelled") | [✅ ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/78675 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/66342 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/26602 "Built successfully") | [✅ ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/12523 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/26512 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/13538 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/28189 "Built successfully") | [✅ ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/73357 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/980 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/28129 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/32819 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

smfr reviewed (approved):
Should this be tested in a WPT?

nt1m committed:
color-scheme should not propagate to viewport background if set on <body>
https://bugs.webkit.org/show_bug.cgi?id=243757

Reviewed by Simon Fraser.

Change to web compatible behavior where setting the color-scheme on <body> no longer propagates it to <html> tag for scrollbar and
webpage background purposes. This change puts WebKit in-line with Gecko & Blink.

* LayoutTests/css-dark-mode/color-scheme-scrollbar-expected.txt:
* LayoutTests/css-dark-mode/color-scheme-scrollbar.html:
* Source/WebCore/page/FrameView.cpp:
(WebCore::FrameView::rendererForColorScheme const):

Canonical link: https://commits.webkit.org/253565@main
